File: zmq/subscribe.rb
#!/usr/bin/env ruby

require 'ffi-rzmq'
require './const'

context = ZMQ::Context.new
s = context.socket(ZMQ::SUB)                # create a subscriber socket
p = "tcp://#{HOST}:#{PORT}"                 # how and where to communicate
s.connect(p)                                # connect to the server
s.setsockopt(ZMQ::SUBSCRIBE, "TIME")        # subscribe to TIME messages

5.times do                  # Five iterations
  time = ""
  s.recv_string(time)       # receive a message
  puts(time)
end

File: zmq/subscribe.py
#!/usr/bin/env python3

import zmq
from const import *

context = zmq.Context()
s = context.socket(zmq.SUB)                 # create a subscriber socket
p = f"tcp://{HOST}:{PORT}"                  # how and where to communicate
s.connect(p)                                # connect to the server
s.setsockopt_string(zmq.SUBSCRIBE, "TIME")  # subscribe to TIME messages

for i in range(5):          # Five iterations
  time = s.recv_string()    # receive a message
  print(time)
